src/core/manager.c

```diff
@@ -4985,6 +5042,98 @@ static int manager_dispatch_handoff_timestamp_fd(sd_event_source *source, int fd
         return 0;
 }
 
+static int manager_dispatch_pidref_fd(sd_event_source *source, int fd, uint32_t revents, void *userdata) {
+        Manager *m = ASSERT_PTR(userdata);
+        _cleanup_(pidref_done) PidRef child_pidref = PIDREF_NULL;
+        CMSG_BUFFER_TYPE(CMSG_SPACE(sizeof(struct ucred)) + CMSG_SPACE(sizeof(int)) * 2) control;
+        struct msghdr msghdr = {
+                .msg_iov = &IOVEC_MAKE(&child_pidref.pid, sizeof(child_pidref.pid)),
+                .msg_iovlen = 1,
+                .msg_control = &control,
+                .msg_controllen = sizeof(control),
+        };
+        ssize_t n;
+        int r;
+
+        assert(source);
+
+        n = recvmsg_safe(m->pidref_fds[0], &msghdr, MSG_DONTWAIT|MSG_CMSG_CLOEXEC|MSG_TRUNC);
+        if (ERRNO_IS_NEG_TRANSIENT(n))
+                return 0; /* Spurious wakeup, try again */
+        if (n == -EXFULL) {
+                log_warning("Got message with truncated control, ignoring.");
+                return 0;
+        }
+        if (n < 0)
+                return log_error_errno(n, "Failed to receive pidref message: %m");
+
+        if (msghdr.msg_flags & MSG_TRUNC) {
```

Comment:
recvmsg_safe() would emit an error if truncated, hence this is redundant (while at it, could you also update _dispatch_handoff_timestamp_fd?